java/clients/src/main/java/sleeper/clients/status/report/statestore/QueryStateStoreCommitterLogs.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.clients.status.report.statestore;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.awssdk.services.cloudwatchlogs.model.GetQueryResultsResponse;
+import software.amazon.awssdk.services.cloudwatchlogs.model.QueryStatus;
+
+import sleeper.configuration.properties.instance.InstanceProperties;
+import sleeper.core.util.PollWithRetries;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toUnmodifiableList;
+import static sleeper.configuration.properties.instance.CdkDefinedInstanceProperty.STATESTORE_COMMITTER_LOG_GROUP;
+
+public class QueryStateStoreCommitterLogs {
+    public static final Logger LOGGER = LoggerFactory.getLogger(QueryStateStoreCommitterLogs.class);
+
+    private final InstanceProperties instanceProperties;
+    private final CloudWatchLogsClient cloudWatch;
+
+    public QueryStateStoreCommitterLogs(InstanceProperties instanceProperties, CloudWatchLogsClient cloudWatch) {
+        this.instanceProperties = instanceProperties;
+        this.cloudWatch = cloudWatch;
+    }
+
+    public List<StateStoreCommitterLogEntry> getLogsInPeriod(Instant startTime, Instant endTime) {
+        String logGroupName = instanceProperties.get(STATESTORE_COMMITTER_LOG_GROUP);
+        LOGGER.info("Submitting logs query for log group {} starting at time {}", logGroupName, startTime);
+        String queryId = cloudWatch.startQuery(builder -> builder
+                .logGroupName(logGroupName)
+                .startTime(startTime.getEpochSecond())
+                .endTime(endTime.getEpochSecond())
+                .limit(10000)
```

Comment:
Limit is good and necessary, but do we need additional capacity to handle if the amount of entries for this window exceeds our limit?

Reply:
That's a good point. It looks like the CloudWatch API doesn't have specific support for paging. 10,000 is just the highest limit that CloudWatch will allow.

Say we had 8,500 commits, that would result in 8,500 + 850 x 2 = 10,200 log entries that would match this query, so we'd lose 200 of them.

This shouldn't be a problem unless we want more than 8,000 commits in one test. If we ran a test that made 10,000 commits and waited for them to finish, it would just time out because it would never be able to find all of those commits.

We could set up a way to page through the results, but we'd risk new logs coming in as we page through. It seems to take about 2 minutes for logs to be available in CloudWatch, so we'd have to wait quite a long time to be confident that it's settled enough for us to read through everything.

For now I'll just add a comment explaining the problem.

Reply:
Comment in code fine for now, think future possible avenue of improvement in a seperate ticket. Approving PR.